modules/core/src/passes/screen-pass.js

```diff
@@ -0,0 +1,47 @@
+//
+// A base render pass.
+//
+// Attribution: This class and the multipass system were inspired by
+// the THREE.js EffectComposer and *Pass classes
+
+import {withParameters} from '@luma.gl/core';
+import Pass from './pass';
+
+export default class ScreenPass extends Pass {
+  render(params) {
+    const gl = this.gl;
+
+    withParameters(gl, {framebuffer: params.outputBuffer, clearColor: [0, 0, 0, 0]}, () =>
+      this._renderPass(gl, params)
+    );
+  }
+
+  cleanup() {
```

Comment:
- A pass should always be "deleted" if it's never recycled.
- From resource management perspective, whoever created the model should be deleting them. Since `model` is provided externally to the `ScreenPass`, deleting it here can cause unexpected side effects.

Reply:
> * A pass should always be "deleted" if it's never recycled.

Do you mean we need change the cleanup() function to delete() or something else?

> * From resource management perspective, whoever created the model should be deleting them. Since `model` is provided externally to the `ScreenPass`, deleting it here can cause unexpected side effects.

Maybe "whoever owns the model should be deleting them" more clearer? Although PostProcessEffect creates the model, but the actual ownership is already transferred to ScreenPass class, from this perspective deleting model here is a good fit.

Reply:
> the actual ownership is already transferred to ScreenPass class

There is no "transfer" happening, you **think** `ScreenPass` owns the model because you **know** what's going on. This is very dangerous when a different maintainer take over this part of the code, because nothing inside the `ScreenPass` class indicates what is expected in `props`. If you want `ScreenPass` to own the model, create it inside the `ScreenPass` constructor.